src/hypothesis/internal/conjecture/engine.py

```diff
@@ -393,8 +394,9 @@ def run(self):
                 self._run()
             except RunIsComplete:
                 pass
-            for v in self.interesting_examples.values():
-                self.debug_data(v)
+            if self.settings.verbosity >= Verbosity.debug:
+                for v in self.interesting_examples.values():
+                    self.debug_data(v)
```

Comment:
While we’re here, why not move the `self.debug()` call on the second line under this `if` statement?

(If you follow it down the same check happens, just several function calls later.)

I think it might be slightly neater here, but I don’t care too much.